clustermesh-apiserver/clustermesh/root.go

```diff
@@ -363,9 +394,9 @@ func startServer(
 	}
 
 	ctx := context.Background()
-	go synchronize(ctx, resources.CiliumIdentities, newIdentitySynchronizer(ctx, cinfo, backend, factory))
-	go synchronize(ctx, resources.CiliumNodes, newNodeSynchronizer(ctx, cinfo, backend, factory))
-	go synchronize(ctx, resources.CiliumSlimEndpoints, newEndpointSynchronizer(ctx, cinfo, backend, factory))
+	go synchronize(ctx, resources.CiliumIdentities, newIdentitySynchronizer(ctx, cinfo, backend, factory, func(_ context.Context) { syncState.identities = true }))
```

Comment:
I'd suggest extending `resourceSyncState` with the appropriate `func(_ context.Context)` method to configure it, so that it is both thread safe (either using locks or `atomic.Bool`) and avoids the need of the lambdas here. Another possibility could be just using an atomic counter, although at the cost of losing information about which type of resources synchronized (which I wonder whether we really need).